aptos-move/aptos-aggregator/src/tests/types.rs

```diff
@@ -119,26 +120,19 @@ impl TDelayedFieldView for FakeAggregatorView {
         DelayedFieldID::new_with_width(id, width)
     }
 
-    fn validate_and_convert_delayed_field_id(
-        &self,
-        id: u64,
-    ) -> Result<Self::Identifier, PanicError> {
-        if id < self.start_counter as u64 {
-            return Err(code_invariant_error(format!(
-                "Invalid delayed field id: {}, we've started from {}",
-                id, self.start_counter
-            )));
-        }
+    fn validate_delayed_field_id(&self, id: &Self::Identifier) -> Result<(), PanicError> {
+        let id = id.extract_unique_index();
+        let current_counter = *self.counter.borrow();
 
-        let current = *self.counter.borrow();
-        if id > current as u64 {
+        // We increment counter before we create an identifier from it, so
+        // its value must be at most the current value.
+        if id < self.start_counter || id > current_counter {
```

Comment:
I still think that if the counter is at a given value, the ID we are validating can't be equal to it, since when the ID is returned, the counter is already incremented (https://github.com/aptos-labs/aptos-core/blob/183c6c8339ae2a6009228d1f1f9abd55aca1b947/aptos-move/block-executor/src/view.rs#L1723).
So it should be id >= current_counter, we can do the same in view.rs (@igor-aptos to also confirm?).

Reply:
Hm, I completely missed that we read it before the increment, somehow